consul/kv_consul.go

```diff
@@ -286,19 +368,30 @@ func (kv *consulKV) Create(
 		kvPair.Action = kvdb.KVCreate
 		if ttl > 0 {
 			var ok bool
-			ok, _, err = kv.client.KV().Acquire(sessionPair, nil)
-			if ok && err == nil {
-				return kvPair, err
-			}
-			if _, err := kv.client.Session().Destroy(sessionPair.Session, nil); err != nil {
-				logrus.Error(err)
-			}
-			if _, err := kv.Delete(key); err != nil {
-				logrus.Error(err)
-			}
-			if err != nil {
-				return nil, err
+
+			for range myParams.machines {
+				ok, _, err = kv.client.KV().Acquire(sessionPair, nil)
+				if ok && err == nil {
+					return kvPair, err
+				}
+				if _, err := kv.client.Session().Destroy(sessionPair.Session, nil); err != nil {
+					logrus.Error(err)
+				}
+				if _, err := kv.Delete(key); err != nil {
+					logrus.Error(err)
+				}
+				if err != nil {
```

Comment:
To reduce code duplication we can add a new function retryOp, which takes in a callback function as an input which will invoke consul APIs respective to each API here.

Reply:
perhaps for next pr. this pr was only to get failover working